### PR TITLE
Add Snowflake connector in Preview mode and skeletton

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

This PRs adds the skeleton of the Snowflake connector & displays it as preview mode on the UI. 
It introduces a feature flag to prepare its rollout. 

Goal is to ship the skeleton as small as possible so  we can // work (and PR review is easy)
- The skeleton of the connector was added, methods are not yet implemented. 
- Regarding models, the strict minimum is in this PR (a model for the connector configuration).
- The worker boilerplate was not added yet. 

### Changes on the UI

<kbd>
Screenshot 2024-09-24 at 21 01 59
</kbd>

If a user clicks on this, it will display a modal saying to contact us to get more info about this connector. 

## Risk

Unused connector at the moment so not very risky. 
If needed both front and connector could be rolled back

## Deploy Plan

- Merge. 
- Run migration 15 on connectors. 
- Deploy connectors. 
- Deploy front.

### CODE REVIEW

File: types/src/front/data_source.ts [hunk 1]
```diff
@@ -121,37 +124,3 @@ export function isDataSourceNameValid(name: string): Result<void, string> {
 
   return new Ok(undefined);
 }
-
-export function isConnectorProviderAllowed(
```
Comment (PopDaph) [inline types/src/front/data_source.ts:125]: This was a duplicate of the `isConnectorProviderAllowedForPlan` method, I removed it.

**Comment by spolu** [comment]
:fire:

File: connectors/migrations/db/migration_15.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
+-- Migration created on Sep 24, 2024
+CREATE TABLE IF NOT EXISTS "snowflake_configurations" (
+    "id"  SERIAL,
+    "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL,
+    "connectorId" INTEGER REFERENCES "connectors" ("id") ON DELETE SET NULL ON UPDATE CASCADE,
```
Comment (fontanierh) [inline connectors/migrations/db/migration_15.sql:6]: should it be non-nullable and set to delete cascade instead ?

File: connectors/src/lib/models/snowflake.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      defaultValue: DataTypes.NOW,
+    },
+  },
+  {
+    sequelize: sequelizeConnection,
+    modelName: "snowflake_configurations",
+    indexes: [{ fields: ["connectorId"], unique: true }],
+  }
+);
+ConnectorModel.hasOne(SnowflakeConfigurationModel);
```
Comment (fontanierh) [inline connectors/src/lib/models/snowflake.ts:48]: IMO should be set to non null (and if not, should be reflected in the type)

**Review by fontanierh — APPROVED** [review]
🔥 LGTM
just one small nit

File: connectors/migrations/db/migration_15.sql (hunk 2 above)
Comment (PopDaph) [reply connectors/migrations/db/migration_15.sql]: Def right, I added a commit!
I put it in restrict since we decided to stop relying on cascade.

**Pull request merged**